lib/auth/clt.go

```diff
@@ -345,29 +345,40 @@ func (c *Client) CompareAndSwapCertAuthority(new, existing types.CertAuthority)
 }
 
 // GetCertAuthorities returns a list of certificate authorities
-func (c *Client) GetCertAuthorities(ctx context.Context, caType types.CertAuthType, loadKeys bool, opts ...services.MarshalOption) ([]types.CertAuthority, error) {
+func (c *Client) GetCertAuthorities(ctx context.Context, caType types.CertAuthType, loadKeys bool) ([]types.CertAuthority, error) {
 	if err := caType.Check(); err != nil {
 		return nil, trace.Wrap(err)
 	}
-	out, err := c.Get(ctx, c.Endpoint("authorities", string(caType)), url.Values{
-		"load_keys": []string{fmt.Sprintf("%t", loadKeys)},
-	})
-	if err != nil {
-		return nil, trace.Wrap(err)
-	}
-	var items []json.RawMessage
-	if err := json.Unmarshal(out.Bytes(), &items); err != nil {
-		return nil, err
-	}
-	re := make([]types.CertAuthority, len(items))
-	for i, raw := range items {
-		ca, err := services.UnmarshalCertAuthority(raw)
+
+	cas, err := c.APIClient.GetCertAuthorities(ctx, caType, loadKeys)
+	switch {
+	case err == nil:
+		return cas, nil
+	case trace.IsNotImplemented(err):
+		resp, err := c.Get(ctx, c.Endpoint("authorities", string(caType)), url.Values{
```

Comment:
Nit: I've often found myself creating a method/function for the old code and placing it in `httpfallback.go` and then calling that from `clt.go` rather than inlining the legacy code. I feel like this makes it a bit easier to scan through methods which are likely queued for deletion in a new Teleport release by just opening up `httpfallback.go`. This isn't a strong opinion, just a thought :')

Reply:
I was mostly just following the pattern I'd noticed previously - given the fact there's currently no methods there - maybe we should just go ahead and delete `httpfallback.go` and just move forward with keeping it inline ?